core/src/labelled.rs

```diff
@@ -617,6 +617,31 @@ where
     }
 }
 
+/// A generic representation for a `Result`.  This is not simply a `Result` to provide
+/// disambiguation between conversion to `Repr` and `transmogrification()` between `Repr`s.
```

Comment:
Can you detail what conflict you run into without this?

Reply:
There's a existing (and fairly critical) conversion for `T -> T::Repr` and I was adding a conversion from `Result::Repr -> Result::Repr`.  If `Result::Repr` was `Result` then there would be no way to disambiguate between these two conversions.